.github/workflows/cypress-workflow.yml

```diff
@@ -0,0 +1,129 @@
+name: E2E Cypress tests
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+jobs:
+  tests:
+    name: Run Cypress E2E tests
+    runs-on: ubuntu-latest
+    env:
+      # prevents extra Cypress installation progress messages
+      CI: 1
+      # avoid warnings like "tput: No value for $TERM and no -T specified"
+      TERM: xterm
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          # TODO: Parse this from security plugin
+          java-version: 13
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          path: security
+          repository: opendistro-for-elasticsearch/security
+      - name: Checkout Security Kibana plugin
+        uses: actions/checkout@v2
+        with:
+          path: security-kibana-plugin
+      - name: Get Kibana version
+        id: kibana_version
+        run: |
+          echo "::set-output name=kibana_version::$(node -p "(require('./security-kibana-plugin/package.json').kibana.version).match(/[.0-9]+/)[0]")"
+      - name: Get Opendistro version
+        id: opendistro_version
+        run: |
+          echo "::set-output name=opendistro_version::$(node -p "(require('./security-kibana-plugin/package.json').version).match(/[.0-9]+/)[0]")"
+      - name: Get Cypress version
+        id: cypress_version
+        run: |
+          echo "::set-output name=cypress_version::$(node -p "(require('./security-kibana-plugin/package.json').devDependencies.cypress).match(/[.0-9]+/)[0]")"
+      - name: Run elasticsearch with plugin
+        run: |
+          wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-${{ steps.kibana_version.outputs.kibana_version }}-darwin-x86_64.tar.gz
+          tar -xzf elasticsearch-oss-${{ steps.kibana_version.outputs.kibana_version }}-darwin-x86_64.tar.gz
+          cd elasticsearch-${{ steps.kibana_version.outputs.kibana_version }}/
+          bin/elasticsearch-plugin install -b https://d3g5vo6xdbdb9a.cloudfront.net/downloads/elasticsearch-plugins/opendistro-security/opendistro_security-${{ steps.opendistro_version.outputs.opendistro_version }}.zip
+          bash plugins/opendistro_security/tools/install_demo_configuration.sh -y -i
+          bin/elasticsearch &
+          sleep 60
```

Comment:
shall we try to call ES API to verify ES is up, instead of wait for 60 seconds?

Reply:
I have a curl after this line to check the status.

Originally, I tried sth like this "           # timeout 300 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' https://localhost:9200 -u admin:admin -k)
" != "200" ]]; do sleep 5; done'"

I tested locally but once running in the workflow, it will fail. Not sure why it happened. Thus I just make it sleep.

Reply:
when the server is not up, the curl command will return a non-zero value, so your check won't work

Reply:
Added back the timeout statement with some tuning of timeout and sleep. The sleep runs parallel with the es start cmd and thus there is not much waste of waiting.

https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/runs/607525181?check_suite_focus=true